crates/lfan/src/cache/ttl_cache.rs

```diff
@@ -0,0 +1,146 @@
+use std::hash::Hash;
+use std::{borrow::Borrow, time::Instant};
+use std::{collections::HashMap, time::Duration};
+
+use super::{
+    cache::Cache,
+    policy::{EvictionPolicy, InsertionPolicy},
+};
+
+struct CacheItem<V> {
+    item: V,
+    last_seen: Instant,
+}
+
+impl<V> CacheItem<V> {
+    pub fn is_expired(&self, ttl: &Duration) -> bool {
+        let now = Instant::now();
+        &now.duration_since(self.last_seen) > ttl
+    }
+}
+
+impl<V> From<V> for CacheItem<V> {
+    fn from(item: V) -> Self {
+        CacheItem {
+            item,
+            last_seen: Instant::now(),
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct TTLCache<K, V, IP, EP>
+where
+    K: Hash + Eq,
+    IP: Default + InsertionPolicy<K>,
+    EP: Default + EvictionPolicy<K>,
+{
+    data: HashMap<K, CacheItem<V>>,
+
+    ttl: Duration,
+
+    insertion_policy: IP,
+    eviction_policy: EP,
+
+    maximum_size: usize,
+}
+
+impl<K, V, IP, EP> TTLCache<K, V, IP, EP>
+where
+    K: Hash + Eq + std::fmt::Debug,
+    IP: Default + InsertionPolicy<K>,
+    EP: Default + EvictionPolicy<K>,
+{
+    pub fn new(maximum_size: usize, ttl: Duration) -> Self {
+        Self {
+            data: HashMap::new(),
+            ttl,
+            insertion_policy: IP::default(),
+            eviction_policy: EP::default(),
+            maximum_size,
+        }
+    }
+}
+
+impl<K, V, IP, EP> Cache for TTLCache<K, V, IP, EP>
+where
+    K: Hash + Eq + std::fmt::Debug,
+    IP: Default + InsertionPolicy<K>,
+    EP: Default + EvictionPolicy<K>,
+{
+    type Key = K;
+    type Value = V;
+
+    fn insert(&mut self, key: K, value: V) -> (bool, Option<V>) {
```

Comment:
I don't think it is straight forward to understand that the return value represents (inserted?, evicted value). Can you please add a doc comment?

Reply:
Will do!